functions-and-scope.js

```diff
@@ -1,83 +1,223 @@
 // * Hoe kan ik iedere waarde van de array checken op deze conditie?
 // * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet,
+// ik dit ergens kan bijhouden?
 // Log het antwoord in de terminal.
 
-// ---- Verwachte uitkomst: 6
+    let numberOfCumLaude = 0;
+    for (let i = 0; i < grades.length; i++) {
+        if (grades[i] >= 8) {
+            numberOfCumLaude++;
+        }
+    }
 
+console.log("\nOpdracht 1a: " + numberOfCumLaude);
+
+    // ---- Verwachte uitkomst: 6
 
 /*  1b: Omschrijven tot een herbruikbare functie   */
-// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
-// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades)
+// en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken,
+// zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+function cumLaude (grades) {
+    let numberOfCumLaude = 0;
+    for (let i = 0; i < grades.length; i++) {
+        if (grades[i] >= 8) {
+            numberOfCumLaude++;
+        }
+    }
+    return numberOfCumLaude;
+}
+
+console.log("\nOpdracht 1b: ");
+console.log("cumLaude(grades): " + cumLaude(grades));
+console.log("cumLaude([6, 4, 5]): " + cumLaude([6, 4, 5]));
+console.log("cumLaude([8, 9, 4, 6, 10]): " + cumLaude([8, 9, 4, 6, 10]));
+
 // ---- Verwachte uitkomsten:
 // cumLaude(grades) geeft 6
 // cumLaude([6, 4, 5]) geeft 0
 // cumLaude([8, 9, 4, 6, 10]) geeft 3
 
-
-
-
 /* Opdracht  2: Gemiddeld cijfer */
 
 /* 2a: Script schrijven  */
-// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is,
+// maar we beginnen met de grades array van hierboven.
 // Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe wordt een gemiddelde berekend?
 // * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
 // * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
 // Log het antwoord in de terminal.
 
+    let average;
+    let sumGrades = 0;
+    for (let i = 0; i < grades.length; i++){
+        sumGrades = sumGrades + grades[i];
```

Comment:
Dit kun je wat korter schrijven
```suggestion
        sumGrades += grades[i];
```